pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -350,20 +379,40 @@ pub mod pallet {
 							u32::from_be_bytes(bytes).try_into().map_err(|_| {
 								DispatchError::Other("Expect: usize in wasm is always ge u32")
 							})
-						})?;
-
-					let source_address =
-						Pallet::<T>::try_range(slice_ref, length_source_address, |source_chain| {
-							Ok(source_chain.to_vec())
-						})?;
-
-					let origin_msg = Pallet::<T>::try_range(slice_ref, slice_ref.len(), |msg| {
-						BoundedVec::try_from(msg.to_vec()).map_err(|_| {
-							DispatchError::Other(
-								"Remaining bytes smaller vector in the first place. qed.",
-							)
-						})
-					})?;
+						},
+					)?;
+
+					let source_address = Pallet::<T>::try_range(
+						slice_ref,
+						length_source_address,
+						Error::<T>::from(MalformedSourceAddress),
+						|source_address| {
+							// NOTE: Axelar simply provides the hexadecimal string of an EVM
+							// address as the `sourceAddress` argument.       Solidity does on the
+							// other side recognize the hex-encoding and encode the hex bytes to
+							//       utf-8 bytes.
+							//
+							//       Hence, we are reverting this process here.
```

Comment:
Nit: Formatting off.
```suggestion
							// NOTE: Axelar simply provides the hexadecimal string of an EVM
							// address as the `sourceAddress` argument. Solidity does on the
							// other side recognize the hex-encoding and encode the hex bytes to
							// utf-8 bytes.
							//
							// Hence, we are reverting this process here.
```